apps/analysis_framework/mutation.py

```diff
@@ -1,45 +1,48 @@
 import graphene
 
-from analysis_framework.models import AnalysisFramework
-from analysis_framework.serializers import (
-    AnalysisFrameworkSerializer,
-)
-from analysis_framework.schema import AnalysisFrameworkType
+from deep.permissions import AnalysisFrameworkPermissions as AfP
+
 from utils.graphene.mutation import (
     generate_input_type_for_serializer,
     GrapheneMutation,
     DeleteMutation,
 )
 
+from .models import AnalysisFramework
+from .serializers import (
+    AnalysisFrameworkSerializer,
+)
+from .schema import AnalysisFrameworkType
+
 
 AnalysisFrameworkInputType = generate_input_type_for_serializer(
     'AnalysisFrameworkInputType',
     serializer_class=AnalysisFrameworkSerializer
 )
 
 
-class UpdateAnalysisFramework(GrapheneMutation):
+class CreateAnalysisFramework(GrapheneMutation):
     class Arguments:
         data = AnalysisFrameworkInputType(required=True)
-        id = graphene.ID(required=True)
 
+    # output fields
     result = graphene.Field(AnalysisFrameworkType)
     # class vars
     serializer_class = AnalysisFrameworkSerializer
     model = AnalysisFramework
-    permissions = []  # TODO:
+    permissions = []
```

Comment:
Should not we handle permission for creating?

Reply:
AnalysisFramework is a root entity, so creating one doesn't require any permission.